### PR TITLE
[Fixes #158] Make the name of the downloaded file dynamic

### LINKED ISSUE #158: Dynamic name for downloaded file
### Current Behavior
Currently, the name is always test.py for the downloaded file

### Expected behavior/code
It should be saved as per the name provided in the project details

### ⚙️ Environment

- `Python` version(s): 3.7.9
- `OS`: Windows 10

### PR DESCRIPTION
**Pull request by therajtiwari**
## Pull Request

fixes #158 

### Description

The downloaded file now has the name as provided in the Output Filename of the project

### Checklist:
- [x] Followed the guidelines mentioned in our [Contributing document](../CONTRIBUTING.md)
- [x] Ensured that there aren't other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change
- [x] This PR is linked to an [Issue](https://www.github.com/Auto-DL/Generator/issues) and the issue number has been mentioned in the title and body of the PR
- [x] The code is clean and docstrings have been added or updated as required
- [x] Code is linted/formatted locally prior to submission (using `prettier`)

### CODE REVIEW

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 1]
```diff
@@ -2927,7 +2924,10 @@ function Step2() {
     };
     const res = await HomeService.download_code(token, data);
     if (res.status === 200) {
-      fileDownload(res.data, "test.py");
+      //check if a valid name exists else give the default name as "output"
```
Comment (ADI10HERO) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2927]: I personally think this comment isn't necessary, what you're trying to do is pretty much clear from the code :) 
```suggestion
```

Comment (therajtiwari) [reply FrontEndApp/v1-react/src/Components/MainApp/Step2.js]: Deleted it. Thanks for the suggestion :)

**Review by ADI10HERO — COMMENTED** [review]
Hey @therajtiwari, @rj8228 had found some test cases where this code might not give the expected output...
for example name of the file is test.py then your code will append an extra `.py` making it test.py.py right?
Can we have a check to remove any `.` or `.py` or `.[AZaz]*` and then append a final `.py`?

**Comment by therajtiwari** [comment]
> Hey @therajtiwari, @rj8228 had found some test cases where this code might not give the expected output...
> for example name of the file is test.py then your code will append an extra `.py` making it test.py.py right?
> Can we have a check to remove any `.` or `.py` or `.[AZaz]*` and then append a final `.py`?

Added the condition to check it. Please review the same.

**Pull request merged**